port/osx/omrosdump.c

```diff
@@ -190,23 +190,36 @@ coredump_to_file(mach_port_t task_port, pid_t pid)
 
 	seg_file_off = file_off + segment_count * sizeof(struct segment_command_64);
 	for (i = 0; i < segment_count; i++) {
-		vm_offset_t data_read = 0;
-		mach_msg_type_number_t data_size = 0;
-
-		kr = mach_vm_read(task_port, segments[i].vmaddr, segments[i].vmsize, &data_read, &data_size);
-		/* any memory segment that is unreadable will be added to the core as an empty segment */
-		if (KERN_SUCCESS == kr) {
-			if ((0 == data_read) || (0 == data_size)) {
-				written = 0;
+		int buffer_size = 65536;
+		mach_vm_address_t data_read = 0;
+		mach_vm_size_t data_size = 0;
+		mach_vm_size_t bytes_read = 0;
+		uint8_t local_buf[buffer_size];
```

Comment:
Would it work to have a pre-allocated buffer, allocated at startup, rather than assuming there will be enough room on the stack?

Reply:
Yes, that works. Will make the change.